src/query/plan/operator.cpp

```diff
@@ -1986,6 +1986,14 @@ class ExpandAllShortestPathsCursor : public query::plan::Cursor {
         }
 
         if ((current_weight > visited_cost_.at(next_vertex)).ValueBool()) continue;
+
+        // Place destination node on the frame, handle existence flag
+        if (self_.common_.existing_node) {
+          const auto &node = frame[self_.common_.node_symbol];
+          if ((node != TypedValue(next_vertex, pull_memory)).ValueBool()) continue;
```

Comment:
It is a slight change, but you can avoid constructing a `TypedValue` by:
```suggestion
          const auto &node = frame[self_.common_.node_symbol];
          ExpectType(self_.common_.node_symbol, existing_node, TypedValue::Type::Vertex);
          if (node.ValueVertex() != next_vertex) continue;
```
Unless I am missing something, in that case please let me know. Also, please format the code before committing, I made the changes on github, so the formatting might be slightly off.

Reply:
And if my solution works, then please also remove the `pull_memory` variable.